server/mdm/android/service/androidmgmt/google_client.go

```diff
@@ -298,3 +325,56 @@ func getLastPart(ctx context.Context, name string) (string, error) {
 	}
 	return nameParts[len(nameParts)-1], nil
 }
+
+type appNotFoundError struct{}
+
+var _ fleet.NotFoundError = (*appNotFoundError)(nil)
+
+func (p appNotFoundError) Error() string {
+	return "Couldn’t add software. The application ID isn’t available in Play Store. Please find ID on the Play Store and try again."
+}
+
+func (p appNotFoundError) IsNotFound() bool {
+	return true
+}
+
+func (g *GoogleClient) EnterprisesApplications(ctx context.Context, enterpriseName, packageName string) (*androidmanagement.Application, error) {
+	if g == nil || g.mgmt == nil {
+		return nil, errors.New("android management service not initialized")
+	}
```

Comment:
We should either have this check everywhere on the methods of `g` or not at all (I don't think it's necessary, in general we don't have this e.g. on Service or Datastore methods and just leave it to panic if something was not initialized, as it tends to be found in development with tests anyway).